documentation/src/main/asciidoc/cheat-sheets/cache-configuration.adoc

```diff
@@ -0,0 +1,414 @@
+//Community doc attributes
+include::../topics/attributes/community-attributes.adoc[]
+
+= {brandname} cache configuration cheat sheet
+
+Use the following information and configuration examples as a quick reference for creating and configuring {brandname} caches.
+
+== Clustered cache modes
+
+Distributed:: Distributed caches offer linear scalability and efficient memory usage by creating fewer copies of each entry across the cluster.
+Distributed caches store a number of copies for each cache entry that is equal to the *Number of owners* that you configure.
++
+[source,xml]
+----
+<distributed-cache />
+----
+
+Replicated:: Replicated caches offer data guarantees and fault tolerance by creating a copy of all entries on each node in the cluster.
++
+[source,xml]
+----
+<replicated-cache />
+----
+
+== Cluster replication
+
+Synchronous replication:: Helps to keep your data consistent but adds latency to cluster traffic that reduces throughput for cache writes.
++
+[source,xml]
+----
+<distributed-cache mode="SYNC" />
+----
+Asynchronous replication:: Reduces latency and increases the speed of write operations but leads to data inconsistency and provides a lower guarantee against data loss.
++
+[source,xml]
+----
+<distributed-cache mode="ASYNC" />
+----
+
+== Cache configuration options
+
+.Encoding
+A format, identified by a media type, that {brandname} uses to store keys and values.
+
+TIP: {brandname} recommends the `application/x-protostream` media type. Protobuf gives you excellent performance as well as interoperability between client applications in different programming languages for both Hot Rod and REST endpoints.
+
+[source,xml]
+----
+<distributed-cache mode="SYNC">
+  <encoding>
+    <key media-type="application/x-protostream"/>
+    <value media-type="application/x-protostream"/>
+  </encoding>
+</distributed-cache>
+----
+
+.Expiration
+Expiration removes entries based on a lifespan or maximum idle time. Expiration takes effect across {brandname} clusters.
+
+Lifespan:: The maximum amount of time, in milliseconds, that cache entries can exist. If you set the value to -1, entries never expire.
+
+Max-idle:: The maximum amount of time, in milliseconds, that cache entries can remain idle. If no operations are performed on entries within the maximum idle time, the entries expire across the cluster. If you set the value to -1, entries never expire.
+
+TIP: Expiration with maximum idle time causes performance issues for `get()` update all instance, lifespan is better if very close.
```

Comment:
@dvagnero in case of max-idle each get need to check whether the last access is longer ago (if the local one is already expired) as it could be that another node might had been accessed.
After the get you need to sync the last-access-date. AFAIK the primary owner will be the source of truth.
As you see you need some internal communication which impact performance.
For lifespan the time of creation (or update) via put is used to compare it with the lifespan, that means only local checks for get()

Reply:
As Tip I would add that setting max-idle close to lifespan has only performance drawbacks and makes no sense. I would suppose that lifespan should be 2*max-idle at minimum